src/services/useDaoMetadata.ts

```diff
@@ -0,0 +1,43 @@
+import fetchJson from '@/utils/fetchJson'
+import { useState, useCallback } from 'react'
+
+type DaoMetadata = {
+  data: any
+  fetch: () => void
+  loading: boolean
+  errors: any
+}
+
+/**
+ * Managing DAO metadata interaction abstracting state managment
+ */
+const useDatoMetadata = (
+  daoAddress: string | string[] | undefined
+): DaoMetadata => {
+  const baseUrl = 'https://data.daohaus.club/dao'
+  const api = useCallback(
```

Comment:
@daniel-ivanco Can you enlighten me why we need useCallback. I am not used to the useCallback pattern.

Reply:
I'd still like to understand this pattern so keeping this open.